tests/static-analysis/assert-psalm-notRedundant.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Webmozart\Assert\Tests\StaticAnalysis;
+
+use Webmozart\Assert\Assert;
+
+/**
+ * @psalm-pure
+ *
+ * @param mixed $value
+ *
+ * @see https://github.com/webmozart/assert/pull/160#issuecomment-564491986
+ * @see https://github.com/vimeo/psalm/commit/4b715cdbffea19a7eab6f72482027d2dd358aab2
+ * @see https://github.com/vimeo/psalm/issues/2456
+ */
+function stringWillNotBeRedundantIfAssertingAndNotUsingEither($value): bool
```

Comment:
waaaait... why do we care?

Reply:
or is this a pre-existing test that was moved? (don't move too much around next time 😱)

Reply:
It was moved to comply with my mental framework: if something need to be checked that is fundamental to psalm properties not to the properties of the `Assert` classes - it should not be mixed and goes to its own `assert-psalm-xxx` files.